components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -29,6 +34,9 @@ open class LoginsStorage {
             try! LoginsStoreError.unwrap({ err in
                 sync15_passwords_state_destroy(raw, err)
             })
+            self.interrupt_handle_lock.lock()
```

Comment:
This won't ever deadlock waiting for something else in the `queue`, right? Since `interrupt` isn't called in a `queue.sync`, and `doOpen` is never called reentrantly?

Reply:
Exactly.